### PR TITLE
StandaloneMmPkg: Standalone MM Lockbox Driver

### PR DESCRIPTION
**Pull request by xieyuanh**
# Description

The Lockbox Driver allows sensitive data to be securely stored in a designated area, thus protected against unauthorized access.

For DXE drivers use lockbox APIs via a communication mechanism triggering an SMI, it's must to have the corresponding SMI handler pre-installed for interrupt management. To ensure orderly operations and proper notification, besides specified the guid in the [Depex] section of the .inf file. The installation of smi handler, along with the LockBox protocol marked by gEfiLockBoxProtocolGuid, must be informed to the DXE driver. 

This patch series introduces a Standalone MM Lockbox Driver. It begins with the registration of a communication handler and a notification for the SMM Ready To Lock Protocol, which ensures a secure transition to the lock phase.

This series works in conjunction with the StandaloneMmPkg: Add LockBox Dependency Library, found here: https://github.com/tianocore/edk2/pull/5940. This library is essential for standalone MM environments where gBS is unavailable, indicating that the LockBox API is ready for use.

- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.
## How This Was Tested
The testing on Intel's next-generation server processor platform. 

## Integration Instructions
N/A

### CODE REVIEW

File: MdeModulePkg/Library/SmmLockBoxLib/SmmLockBoxStandaloneMmLib.inf [hunk 1]
```diff
@@ -38,6 +38,7 @@
   MmServicesTableLib
   BaseLib
   DebugLib
+  StandaloneMmDriverEntryPoint
```
Comment (niruiyu) [inline MdeModulePkg/Library/SmmLockBoxLib/SmmLockBoxStandaloneMmLib.inf:41]: Why a lib needs to depend on DriverEntryPoint?

File: MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBoxCommon.h [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+#include <Library/BaseMemoryLib.h>
+#include <Library/DebugLib.h>
+#include <Library/LockBoxLib.h>
+
+#include <Protocol/SmmReadyToLock.h>
+#include <Protocol/SmmCommunication.h>
+#include <Protocol/LockBox.h>
+#include <Guid/SmmLockBox.h>
+
+extern BOOLEAN  mLocked;
```
Comment (niruiyu) [inline MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBoxCommon.h:29]: I feel "extern BOOLEAN mLocked" is not needed.

File: MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBoxStandaloneMm.c
Comment (niruiyu) [inline MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBoxStandaloneMm.c:1]: I am a bit confused. The SmmLockBoxStandaloneMm.c is a new file. Why the commit message says there is no functionality?

File: MdeModulePkg/Library/SmmLockBoxLib/SmmLockBoxStandaloneMmLib.inf (hunk 1 above)
Comment (xieyuanh) [reply MdeModulePkg/Library/SmmLockBoxLib/SmmLockBoxStandaloneMmLib.inf]: Hi Ray Thanks I've remove it in the lastest version.

File: MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBoxCommon.h (hunk 2 above)
Comment (xieyuanh) [reply MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBoxCommon.h]: Hi Ray Thanks I've remove it in the lastest version.

File: MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBoxStandaloneMm.c
Comment (xieyuanh) [reply MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBoxStandaloneMm.c:1]: SmmLockBoxStandaloneMm.c  has been removed to the third patch.

File: MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBoxStandaloneMm.c [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+  //
+  // Register SMM Ready To Lock Protocol notification
+  //
+  Status = gMmst->MmRegisterProtocolNotify (
+                    &gEfiSmmReadyToLockProtocolGuid,
+                    SmmReadyToLockEventNotify,
+                    &Registration
+                    );
+  ASSERT_EFI_ERROR (Status);
```
Comment (niruiyu) [inline MdeModulePkg/Universal/LockBox/SmmLockBox/SmmLockBoxStandaloneMm.c:82]: sorry for the last minute comments.
Can you please move the two function calls into Common.c?

**Review by niruiyu — APPROVED** [review]

**Review by niruiyu — COMMENTED** [review]
only one comment: the commit message of the 3rd commit is not correct. Please correct it.

**Pull request closed without merging**